OpenRA.Mods.Common/Util.cs

```diff
@@ -155,5 +158,64 @@ public static IEnumerable<CPos> RandomWalk(CPos p, MersenneTwister r)
 				yield return p;
 			}
 		}
+
+		public static string FriendlyTypeName(Type t)
+		{
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(HashSet<>))
+				return "Set of {0}".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
+
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Dictionary<,>))
+				return "{0}, {1} pairs".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
+
+			if (t.IsSubclassOf(typeof(Array)))
+				return "Multiple {0}".F(FriendlyTypeName(t.GetElementType()));
+
+			if (t.IsGenericType && t.GetGenericTypeDefinition().GetInterfaces().Any(e => e.IsGenericType && e.GetGenericTypeDefinition() == typeof(IEnumerable<>)))
+				return "Multiple {0}".F(FriendlyTypeName(t.GetGenericArguments().First()));
```

Comment:
"Collection of {0}" would be better here: it agrees better with "Set of {0}" above.  You can also combine this with the array check above.